data/core/macros/utils.cfg

```diff
@@ -578,6 +578,10 @@
             name=attack end
             delayed_variable_substitution=yes
 
+            [filter]
+                {FILTER}
+            [/filter]
```

Comment:
What guarantees that this `attack end` event will fire at the end of the attack that triggered the outside `attack` event? If {FILTER} is `id=Konrad` it may be fine, but {FILTER} could also be something like `side,hitpoints=1,1-20` which could have different values at `attack` and `attack end`. I think the inner event needs to fire at the end of the very same attack, so the filter should be removed and something else done to prevent the bug you ran into. Probably something along the lines of making  [unstore_unit] conditional on [have_unit] id=$unit.id [/have_unit], but don't catch me on it.